
# Code Pipeline for RNA-Seq Mapping and Read Quantification in Hi-A Corn

Welcome to this repository! This collection contains the scripts and workflows used the Hi-A Corn Transcriptomics Project. These analyses were conducted on the Texas A&M High Performance Research Computing (HPRC) cluster.

## Overview
This repository provides a step-by-step workflow for processing RNA-Seq data, from raw read quality control to reads quantification
## Directory Structure
**01_mapping:** This section covers the alignment of raw reads to the reference genome using STAR.
- *fastqc_terrra.slurm*: Quality check of raw reads.
- *trimmomatic_loop_terra.slurm*: Trimming of adapter sequences.
- *trimmomatic_terra.slurm* (Trimming reads, alternative script)
- *ref_genome_index_star_terra.slurm*: Generation of the STAR reference genome index.
- *mapping_reads_indexed_ref_genome.slurm*: Mapping reads to the indexed reference genome.
- *script_counting_STAR_summary*: Script for summarizing mapping statistics.
    
**02_quantification:** This section contains scripts for counting raw reads and estimating transcript quantification in TPM.     
 - *script_salmon_indexing.slurm*: Creating reference genome index with Salmon.
 - *script_salmon_quantifying.slurm*: Transcript quantification using Salmon.
 - *script_processing_STAR_output_files.R*: Script for merging gene-level reads counts for 18 samples generated by STAR into a single dataset.           
            
## Languages Used
- Bash
- R

